Added space not found documentation (#377)
tarantool/cartridge-java: docs/SpaceNotFound.md

## About problem
If you create spaces only on storages inside a storage role [like this](https://github.com/tarantool/cartridge-java/blob/ba6e8b248b6f448cc9f29e4d660a3d0a2f210ccd/src/test/resources/cartridge/app/roles/api_storage.lua#L12-L220)
or any other way, and you want to get data using router. You will receive an error message like "space not found".  
This happens because router does not have information about this space.  
If you work directly with a storage you won't get this kind of error if spaces was created properly.  
But if you want to use [CRUD](https://github.com/tarantool/crud) to retrieve data from the router, there are some solutions.  

## Migrations
We recommend to set the scheme using migrations, they will be applied not only to storages, but also to routers.  
This way the router will be aware of the current scheme and such errors will not occur.
Migrations will create spaces on both storages and routers.  
But Tarantool 1.x community edition does not support migrations.
They are supported on versions 1.10.x and 2.2.x of [Tarantool Enterprise Edition](https://www.tarantool.io/ru/product/enterprise/)

> and newer. Also, Tarantool CE 2.2.x and newer supports migrations.

Reviewer: I suggest writing simpler. Tarantool CE supports migrations only since 2.2.x. Tarantool EE supports migrations since 1.10.x.

More information about migrations [tarantool/migrations](https://github.com/tarantool/migrations).  

## Proxying ddl.get_schema

> If you do not want to use migrations, there is another solution.

Reviewer: ```suggestion
If you want to create spaces only in storages, there is another solution.
```

Because you can also suggest creating spaces on all instances by modules, like:

With the same code on all roles
```lua
local spaces = require'spaces'

function init()
  -- TODO: master condition
  spaces.create()
end
```
